test/rekt/features/kafkachannel/util.go

```diff
@@ -0,0 +1,192 @@
+func setupResetOffset(f *feature.Feature, name string, offsetTime string) {
+	f.Setup("Install A ResetOffset", resetoffsetresources.Install(name,
+		resetoffsetresources.WithOffsetTime(offsetTime),
+		resetoffsetresources.WithRef(&duckv1.KReference{
+			Kind:       "Subscription",
+			Name:       name,
+			APIVersion: "messaging.knative.dev/v1", // gvr() in knative.dev/eventing/test/rekt/resources/subscription is private
+		}),
+	))
+}
+
+// setupEventsHubSender adds a Setup() to the specified Feature to create an EventsHub Sender to send CloudEvents.
+func setupEventsHubSender(f *feature.Feature, senderName string, senderSink string, event cloudevents.Event, eventCount int) {
+	f.Setup("Install An EventsHub Sender", eventshub.Install(senderName,
+		eventshub.StartSender(senderSink),
+		eventshub.InitialSenderDelay(5*time.Second),
+		eventshub.InputEvent(event),
+		eventshub.SendMultipleEvents(eventCount, 200*time.Millisecond),
+		eventshub.EnableIncrementalId,
+	))
+}
+
+// setupEventsHubReceiver adds a Setup() to the specified Feature to create an EventsHub Receiver to receive CloudEvents.
+func setupEventsHubReceiver(f *feature.Feature, receiverName string) {
+	f.Setup("Install An EventsHub Receiver", eventshub.Install(receiverName,
+		eventshub.StartReceiver,
+		eventshub.EchoEvent,
+	))
+}
+
+//
+// Assert Utilities
+//
+
+// assertKafkaChannelReady adds an Assert() to the specified Feature to verify the KafkaChannel is READY.
+func assertKafkaChannelReady(f *feature.Feature, name string) {
+	f.Assert("KafkaChannel Is Ready", kafkachannelresources.IsReady(name))
+}
+
+// assertSubscriptionReady adds an Assert() to the specified Feature to verify the Subscription is READY.
+func assertSubscriptionReady(f *feature.Feature, name string) {
+	f.Assert("Subscription Is Ready", subscriptionresources.IsReady(name))
+}
+
+// assertResetOffsetSucceeded adds an Assert() to the specified Feature to verify the ResetOffset is SUCCEEDED
+func assertResetOffsetSucceeded(f *feature.Feature, name string) {
+	f.Assert("ResetOffset Is Succeeded", resetoffsetresources.IsSucceeded(name))
+}
+
+// assertEventsReceived adds an Assert() to the specified Feature to verify the expected CloudEvents were received successfully.
+func assertEventsReceived(f *feature.Feature, receiverName string, event cloudevents.Event, eventCount int, startId int, endId int) {
+	matchers := newEventMatcher(event, startId, endId)
+	f.Assert("Events Received", assert.OnStore(receiverName).MatchEvent(matchers).Exact(eventCount))
+}
+
+//
+// CloudEvent Utilities
+//
+
+// newEvent returns a new CloudEvent suitable for testing.
+func newEvent(name string, senderName string) cloudevents.Event {
+	event := cloudevents.NewEvent()
+	event.SetSource(senderName)
+	event.SetType(name + "-type")
+	event.SetSubject(name + "-subject")
+	_ = event.SetData(cloudevents.ApplicationJSON, map[string]string{"name": name})
```

Comment:
I'd rather keep "t" out of newEvent() and instead just return the error.  Originally I had ignored the error because this is just test code that should never fail given it has "mostly" fixed content and should never fail, but... you are right there is still a possibility of it happening.  I'll make changes to return error and check in caller.